### DISCUSSION TITLE
Hope to know more about the visualization of fence diagram

### DISCUSSION THREAD

**Discussion by nnhnfh** [emanuelhuber/RGPR#71] [Q&A]
Dear Mr. Emmanuel Huber,

Thank you so much for creating this package.
Actually this is my first time for me to do processing with GPR data. And i want to visualize my GPR data to fence diagram, but i have problem here.  I want to layout the visualization of my fence diagram which is will be include the information about coordinate and the depth of data. And i want to show the data in specific depth. I did read the source but i could not find the script that i need. Here, I attach the visualization result of the fence diagram that I have created with this script plot3DRGL(mySurvey, addTopo = TRUE)
![plot3d](https://github.com/emanuelhuber/RGPR/assets/156985551/ee4ca9af-de14-4dc5-8084-4bdd602282b8)

Would you mind to explain the arguments in plot3DRGL() in detail way? 
Hoping to hear from you soon!
Best wishes!

Sincerely,
Nove

**Comment by emanuelhuber** [comment]
Hello Nove
Thank you for your request. I will try to provide you a solution this week.
Best,
Emanuel

**Reply by nnhnfh** [reply]
Thank you so much for your response, Mr. Emanuel. I was trying to add the xlim, ylim, and zlim parameters; but it shows the same as the previous picture I sent you. And I hope I can have your response immediately.

Best wish,
Nove

**Reply by emanuelhuber** [reply]
Hello @nnhnfh 

I updated the function `plot3DRGL()` with the option to restrain the data range. This function is still experimental. I will test it tomorrow.

```
plot3DRGL(x, 
                   xlim = c(2626740.0,  2626640.0), 
                   ylim = c(1212680, 1212980),
                   zlim = c(430, 440))

```

Note that you have to provide the coordinates of the 3D bounding box in the regional coordinates of your data (you can check the coordinates with `coord(x)`).

Best,

Emanuel
